web/src/main/java/com/epam/esm/controller/OrderController.java

```diff
@@ -26,32 +28,41 @@ public OrderController(OrderService orderService, OrderModelAssembler orderModel
     @GetMapping(value = "/{id}")
     @PreAuthorize("hasAnyRole('USER','ADMIN')")
     public OrderModel getOrder(@PathVariable("id") Long id) {
-        return orderModelAssembler.toModel(orderService.findById(id)
-                .orElseThrow(() -> new NotFoundException("Order not found with id: " + id)));
+        return orderModelAssembler.toModel(orderService.findById(id));
     }
 
     @GetMapping(value = "/paged")
-    @PreAuthorize("hasAnyRole('USER','ADMIN')")
-    public CollectionModel<OrderModel> getOrdersWithPage(@RequestParam(required = false, defaultValue = "1", name = "page") int page,
-                                                         @RequestParam(required = false, defaultValue = "10", name = "size") int size) {
+    @PreAuthorize("hasRole('ADMIN')")
+    public CollectionModel<OrderModel> getOrdersWithPage(
+            @RequestParam(required = false, defaultValue = "1", name = "page") int page,
+            @RequestParam(required = false, defaultValue = "10", name = "size") int size) {
         List<Order> orders = orderService.findAllWithPage(page, size);
-
         return orderModelAssembler.toCollectionModel(orders, page, size);
     }
 
     @PostMapping
-    @PreAuthorize("hasRole('ADMIN')")
-    public OrderModel createOrder(@RequestBody OrderDTO dto) {
-        Order createdOrder = orderService.create(dto.userId(), dto.giftCertificateId());
-        return orderModelAssembler.toModel(createdOrder);
+    @PreAuthorize("hasAnyRole('ADMIN', 'USER')")
+    public OrderModel createOrder(@RequestBody OrderDTO dto, Authentication authentication) {
+        return orderModelAssembler.toModel(orderService.createOrder(dto, authentication));
     }
 
     @GetMapping(value = "/{userId}/user-orders")
-    @PreAuthorize("hasAnyRole('ADMIN', 'USER')")
-    public CollectionModel<OrderModel> getOrdersByUserIdWithPage(@PathVariable("userId") Long userId,
-                                                                 @RequestParam(required = false, defaultValue = "1", name = "page") int page,
-                                                                 @RequestParam(required = false, defaultValue = "10", name = "size") int size) {
-        List<Order> userOrders = orderService.findOrdersInfoByUserIdWithPage(userId, page, size);
-        return orderModelAssembler.toCollectionModel(userOrders, page, size);
+    @PreAuthorize("isAuthenticated()")
+    public CollectionModel<OrderModel> getOrdersByUserIdWithPage(
```

Comment:
You can use Authentication to get user id if this endpoint is for only one user. You don't need user id in path variable.